docs/mkdocs/labsv2.md

```diff
@@ -0,0 +1,425 @@
+sh-5.1# lsmod | grep kmm   
+kmm_ci_a               16384  0
+sh-5.1# 
+```
+
+## Device Plugins
+
+
+Device plugins play a crucial role in allocating and reporting specific hardware devices to the cluster. In KMM, you can load device plugin images as part of the Module object.
+
+For this example, we will utilize this plugin called [simple-device-plugin](https://github.com/yevgeny-shnaidman/simple-device-plugin) to simulate device plugins. This plugin provides a way to emulate device plugins and test their functionality within the cluster.
+
+It just loads some dummy devices settings in nodes as if they were reported by real hardware and then used by the moduleLoader.
+
+`device_plugin.yaml`
+```yaml
+---
+apiVersion: kmm.sigs.x-k8s.io/v1beta1
+kind: Module
+metadata:
+  name: kmm-ci-a
+  namespace: labsv2 
+spec:
+  devicePlugin:
+    serviceAccountName: labsv2sa
+    container:
+      image: myrepo.xyz/simple-device-plugin:latest
+      imagePullPolicy: Always
+      args:
+      - "-config=config-labs.yaml"
+  moduleLoader:
+    serviceAccountName: labsv2sa
+    container:
+      modprobe:
+        moduleName: kmm-ci-a
+      kernelMappings:
+        - regexp: '^.*'
+          containerImage: myrepo.xyz/kmm-labs:kmm-kmod-5.14.0-284.36.1.el9_2.x86_64-v1.0
+  imageRepoSecret:
+    name: myrepo-pull-secret
+  selector:
+    node-role.kubernetes.io/worker: ""
+
+```
+We can check that our device plugin app has been loaded just describing one of the selected nodes and looking at the resources.
+
+```bash
+$ oc describe node mynode
+...
+Allocated resources:
+  (Total limits may be over 100 percent, i.e., overcommitted.)
+  Resource                   Requests      Limits
+  --------                   --------      ------
+  cpu                        749m (9%)     0 (0%)
+  memory                     4406Mi (29%)  0 (0%)
+  ephemeral-storage          0 (0%)        0 (0%)
+  hugepages-1Gi              0 (0%)        0 (0%)
+  hugepages-2Mi              0 (0%)        0 (0%)
+  dummy/dummyDev             0             0
+  example.com/ptemplate      0             0
+  example.com/simple-device  0             0
+...
+```
+
+##  Seamless upgrade of kernel Module: Orderly process without reboot
+
+To minimize the impact on the workload running in the cluster, kernel module upgrades should be performed on a per-node basis.
+
+First, label the specific node that requires the kernel module upgrade. Use the following format:
+
+```
+kmm.node.kubernetes.io/version-module.<module-namespace>.<module-name>=$moduleVersion
+```
+
+In our case we'll use this example:
+```bash
+oc label node my-node-1 kmm.node.kubernetes.io/version-module.labsv2.kmm-ci-a=0.9
+```
+
+Then we can use a new build based on previous build example by just adapting the ModuleLoader (adding `version` and specific `containerImage` for that version) and reusing existing Configmap for build. Make sure you have deleted any Module object left from past examples and apply following Module:
```

Comment:
I think we should mention that  "ordered upgrade" works only in case "version field was defined in the Module from the very-begining: it won't work on an already existing Module without version field, which will be added during the Module edit